cmake/target_toolchain.cmake

```diff
@@ -1,8 +1,42 @@
-set(CMAKE_SYSTEM_NAME      Generic)
+# No official documentation exists for the "Generic" value, except their wiki.
+#
+# https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/CrossCompiling:
+#   CMAKE_SYSTEM_NAME : this one is mandatory, it is the name of the target
+#   system, i.e. the same as CMAKE_SYSTEM_NAME would have if CMake would run
+#   on the target system.  Typical examples are "Linux" and "Windows". This
+#   variable is used for constructing the file names of the platform files
+#   like Linux.cmake or Windows-gcc.cmake. If your target is an embedded
+#   system without OS set CMAKE_SYSTEM_NAME to "Generic".
+set(CMAKE_SYSTEM_NAME Generic)
+
+# https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html:
+#   The name of the CPU CMake is building for.
+#
+# https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/CrossCompiling:
+#   CMAKE_SYSTEM_PROCESSOR : optional, processor (or hardware) of the
+#   target system. This variable is not used very much except for one
+#   purpose, it is used to load a
+#   CMAKE_SYSTEM_NAME-compiler-CMAKE_SYSTEM_PROCESSOR.cmake file,
+#   which can be used to modify settings like compiler flags etc. for
+#   the target
```

Comment:
I've had a look at how this variable affects the build and I believe it is safe to remove this line.

Then we don't have to have code and docs causing noise in the source.

We do not load "CMAKE_SYSTEM_NAME-compiler-CMAKE_SYSTEM_PROCESSOR.cmake" files in Zephyr.

Reply:
> We do not load "CMAKE_SYSTEM_NAME-compiler-CMAKE_SYSTEM_PROCESSOR.cmake" files in Zephyr.

Yep.

> I've had a look at how this variable affects the build and I believe it is safe to remove this line.

But I see libmetal and and open-amp assumes `CMAKE_SYSTEM_PROCESSOR` (and mentions it their `README.md`):

```
$ rg -l CMAKE_SYSTEM_PROCESSOR
cmake/target_toolchain.cmake
ext/hal/libmetal/libmetal/README.md
ext/lib/ipc/open-amp/open-amp/README.md
ext/hal/libmetal/libmetal/cmake/options.cmake
ext/lib/ipc/open-amp/open-amp/cmake/options.cmake
...

$ rg CMAKE_SYSTEM_PROCESSOR ext/hal/libmetal/
ext/hal/libmetal/libmetal/cmake/options.cmake
22:set (_target "${CMAKE_SYSTEM_NAME}/${CMAKE_SYSTEM_PROCESSOR}")
38:string (TOLOWER ${CMAKE_SYSTEM_PROCESSOR} PROJECT_PROCESSOR)
39:string (TOUPPER ${CMAKE_SYSTEM_PROCESSOR} PROJECT_PROCESSOR_UPPER)

$ rg PROJECT_PROCESSOR ext/hal/libmetal/
ext/hal/libmetal/libmetal/cmake/options.cmake
38:string (TOLOWER ${CMAKE_SYSTEM_PROCESSOR} PROJECT_PROCESSOR)
39:string (TOUPPER ${CMAKE_SYSTEM_PROCESSOR} PROJECT_PROCESSOR_UPPER)

ext/hal/libmetal/libmetal/lib/atomic.h
23:# include <metal/processor/@PROJECT_PROCESSOR@/atomic.h>

ext/hal/libmetal/libmetal/lib/config.h
36:#define METAL_PROCESSOR              "@PROJECT_PROCESSOR@"
37:#define METAL_PROCESSOR_@PROJECT_PROCESSOR_UPPER@

ext/hal/libmetal/libmetal/lib/cpu.h
15:# include <metal/processor/@PROJECT_PROCESSOR@/cpu.h>

ext/hal/libmetal/libmetal/lib/processor/CMakeLists.txt
1:add_subdirectory (${PROJECT_PROCESSOR})
```

Because it does not hurt to keep `CMAKE_SYSTEM_PROCESSOR` and it is official (so other future external components may rely on it), I'd say we should keep it.

Reply:
> I'd say we should keep it.

Agreed.